modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/test/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DefaultDDMFormValuesFactoryTest.java

```diff
@@ -221,6 +222,66 @@ public void testNestedDDMFormFieldValue() {
 			StringPool.BLANK, ageValue.getString(ddmForm.getDefaultLocale()));
 	}
 
+	@Test
+	public void testPopulate() {
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm();
+
+		DDMFormField ddmFormField = DDMFormTestUtil.createDDMFormField(
+			"parentField", null, null, null, false, true, false);
+
+		ddmFormField.addNestedDDMFormField(
+			DDMFormTestUtil.createDDMFormField(
+				"nestedField", null, null, null, false, false, false, null,
+				"true", null));
+
+		ddmForm.addDDMFormField(ddmFormField);
+
+		DefaultDDMFormValuesFactory defaultDDMFormValuesFactory =
+			new DefaultDDMFormValuesFactory(ddmForm);
+
+		DDMFormValues ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(
+			ddmForm);
+
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+				ddmFormField.getName(), "someValue1"));
+
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+				ddmFormField.getName(), "someValue2"));
+
+		defaultDDMFormValuesFactory.populate(ddmFormValues);
+
+		Map<String, List<DDMFormFieldValue>> ddmFormFieldValuesMap =
+			ddmFormValues.getDDMFormFieldValuesMap(false);
+
+		List<DDMFormFieldValue> ddmFormFieldValues = ddmFormFieldValuesMap.get(
+			"parentField");
+
+		_assertDDMFormFieldValue(ddmFormFieldValues.get(0), "someValue1");
+		_assertDDMFormFieldValue(ddmFormFieldValues.get(1), "someValue2");
+	}
+
+	private void _assertDDMFormFieldValue(
+		DDMFormFieldValue ddmFormFieldValue, String expectedValue) {
```

Comment:
@rodrigopaulino can you please check if it’s possible to reuse this method in the existing test `testNestedDDMFormFieldValue()` by adding a new parameter called `expectedNestedValue`? Thanks!

Reply:
I'll check that.